src/setfit/trainer.py

```diff
@@ -195,11 +197,57 @@ def call_model_init(self, params: Dict[str, Any] = None):
 
         return model
 
-    def train(self, trial: Union["optuna.Trial", Dict[str, Any]] = None):
+    def freeze(self):
+        """
+        Freeze SetFitModel's differentiable head.
+        Note: call this function only when using the differentiable head.
```

Comment:
Maybe we could do a small check on the `SetFitModel` and raise a `ValueError` if a user tries to execute this with the wrong model?

Reply:
That's a great idea. Let me add it!

Reply:
Just added it